util/info.c

```diff
@@ -228,29 +229,55 @@ static const char *param_type(enum fi_param_type type)
 	}
 }
 
+static char * get_var_prefix(const char *prov_name)
+{
+	int i;
+	char *prefix;
+
+	if (!prov_name)
+		return NULL;
+	else {
+		if (asprintf(&prefix, "FI_%s", prov_name) < 0) {
+			prefix = NULL;
+			return NULL;
+		}
+		for (i = 0; i < prefix[i]; ++i)
+			prefix[i] = toupper((unsigned char) prefix[i]);
+	}
+
+	return prefix;
+}
+
 static int print_vars(void)
 {
 	int ret, count, i;
 	struct fi_param *params;
 	char delim;
+	char *var_prefix;
 
 	ret = fi_getparams(&params, &count);
 	if (ret)
 		return ret;
 
+	var_prefix = get_var_prefix(hints->fabric_attr->prov_name);
+
 	for (i = 0; i < count; ++i) {
-		printf("# %s: %s\n", params[i].name, param_type(params[i].type));
-		printf("# %s\n", params[i].help_string);
+		if (!(var_prefix) || !strncmp(params[i].name, var_prefix, strlen(var_prefix))) {
```

Comment:
The diff would be simpler if we inverted this if statement and called continue, rather than moving the entire for statement inside an if clause.  I.e.
```
if (var_prefix && strncmp(...))
    continue;

/* original code flows as previously */
```